includes/woopay/class-woopay-scheduler.php

```diff
@@ -117,6 +117,12 @@ public function update_compatibility_and_maybe_show_incompatibility_warning() {
 	public function update_enabled_adapted_extensions( $active_plugins, $adapted_extensions ) {
 		$enabled_adapted_extensions = $this->get_extensions_in_list( $active_plugins, $adapted_extensions );
 
+		if ( count( $enabled_adapted_extensions ) > 0 ) {
```

Comment:
Nit: No changes necessary, but we could also do something like the following:
```php
// ...
$is_first_party_auth_enabled = \WC_Payments_Features::is_woopay_first_party_auth_enabled();
if ( count( $enabled_adapted_extensions ) > 0 && $is_first_party_auth_enabled ) {
  update_option( \WC_Payments_Features::WOOPAY_FIRST_PARTY_AUTH_FLAG_NAME, 0 );
} else if ( count( $enabled_adapted_extensions ) === 0 && ! $is_first_party_auth_enabled ) {
  update_option( \WC_Payments_Features::WOOPAY_FIRST_PARTY_AUTH_FLAG_NAME, 1 );
}
// ...
```

This way, we only update the option if it's necessary. Although, this is not necessary since this block of code only executes when plugins are activated/deactivated.